hazelcast-client-legacy/src/test/java/com/hazelcast/client/map/ClientMapBasicTest.java

```diff
@@ -314,6 +314,72 @@ public void testPutAsyncTTL_afterExpireWhenKeyExists() throws Exception {
     }
 
     @Test
+    public void testSetAsync() throws Exception {
+        IMap<String, String> map = client.getMap(randomString());
+        String key = "Key";
+        String value = "Val";
+
+        Future<Void> result = map.setAsync(key, value);
+
+        result.get();
+        assertEquals(value, map.get(key));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testSetAsync_withKeyNull() throws Exception {
+        IMap<String, String> map = client.getMap(randomString());
+        String val = "Val";
+
+        map.setAsync(null, val);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testSetAsync_withValueNull() throws Exception {
+        IMap<String, String> map = client.getMap(randomString());
+        String key = "key";
+
+        map.setAsync(key, null);
+    }
+
+    @Test
+    public void testSetAsyncTTL() throws Exception {
+        IMap<String, String> map = client.getMap(randomString());
+        String key = "Key";
+        String value = "Val";
+
+        Future<Void> result = map.setAsync(key, value, 5, TimeUnit.MINUTES);
+
+        result.get();
+        assertEquals(value, map.get(key));
+    }
+
+    @Test
+    public void testSetAsyncTTL_afterExpire() throws Exception {
+        IMap<String, String> map = client.getMap(randomString());
+        String key = "Key";
+        String value = "Val";
+
+        Future<Void> result = map.setAsync(key, value, 1, TimeUnit.SECONDS);
+        sleepSeconds(2);
+        result.get();
+        assertEquals(null, map.get(key));
+    }
+
+    @Test
+    public void testSetAsyncTTL_afterExpireWhenKeyExists() throws Exception {
+        IMap<String, String> map = client.getMap(randomString());
+        String key = "Key";
+        String oldValue = "oldValue";
+        String newValue = "Val";
+
+        map.set(key, oldValue);
+        Future<Void> result = map.setAsync(key, newValue, 1, TimeUnit.SECONDS);
+        sleepSeconds(2);
```

Comment:
The sleep is necessary to ensure that the desired number of seconds has elapsed after which the value would be evicted from the Map. I can rework the test such that the test blocks on a latch that is triggered by an eviction listener